presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java

```diff
@@ -139,18 +147,14 @@ private void killContainersReaperContainer()
             }
         }
 
-        private void sleepUntilInterrupted()
+        private void waitUntil(CheckedSupplier<Boolean> completed)
         {
-            try {
-                //noinspection InfiniteLoopStatement
-                while (true) {
-                    Thread.sleep(Long.MAX_VALUE);
-                }
-            }
-            catch (InterruptedException e) {
```

Comment:
What happens now upon ctrl+c?

Reply:
Interrupts don't retry. The process terminates and cleans up properly.

Reply:
Also, I always forget `InturrupedExceptions` are unrelated to SIGINT (ctrl+c) signals. They're just for when the `interrupt()` method on a Thread is called internally by another part of the code, and we're not using it.

Reply:
Thanks for reminding me.